File: PD/SSCMax.java
/*The algorithm search the max longest increasing subsequence of a list
*/ 
public class SSCMax{
	//return a list with all possibilities
	private int[] SSCMaxPD(int[] A, int N){
	int[] c = new int[N];
	for (int m = 0;m < N;m++) {
		c[m] = 1;
		for (int i = m-1;i >= 0 ; i--) {
			if (A[i] <= A[m] && c[i]+1 > c[m]) {
				c[m] = c[i]+1;
			}			
		}
	}
	return c;
	}

}

File: PD/SSCMax.py
#The algorithm search the max longest increasing subsequence of a list

def SSCMax(A,N):
	c = [0 for i in range(N)]
	for m in range(N):
		c[m] = 1
		for i in range(m-1,-1,-1):
			if A[i] <= A[m] and c[i]+1 > c[m]:
				c[m] = c[i]+1
	return c	
